packages/store/src/Types.sol

```diff
@@ -1,84 +1,238 @@
+  BYTES1,
+  BYTES2,
+  BYTES3,
+  BYTES4,
+  BYTES5,
+  BYTES6,
+  BYTES7,
+  BYTES8,
+  BYTES9,
+  BYTES10,
+  BYTES11,
+  BYTES12,
+  BYTES13,
+  BYTES14,
+  BYTES15,
+  BYTES16,
+  BYTES17,
+  BYTES18,
+  BYTES19,
+  BYTES20,
+  BYTES21,
+  BYTES22,
+  BYTES23,
+  BYTES24,
+  BYTES25,
+  BYTES26,
+  BYTES27,
+  BYTES28,
+  BYTES29,
+  BYTES30,
+  BYTES31,
+  BYTES32,
+  BOOL,
+  ADDRESS,
+  UINT8_ARRAY,
+  UINT16_ARRAY,
+  UINT24_ARRAY,
+  UINT32_ARRAY,
+  UINT40_ARRAY,
+  UINT48_ARRAY,
+  UINT56_ARRAY,
+  UINT64_ARRAY,
+  UINT72_ARRAY,
+  UINT80_ARRAY,
+  UINT88_ARRAY,
+  UINT96_ARRAY,
+  UINT104_ARRAY,
+  UINT112_ARRAY,
+  UINT120_ARRAY,
+  UINT128_ARRAY,
+  UINT136_ARRAY,
+  UINT144_ARRAY,
+  UINT152_ARRAY,
+  UINT160_ARRAY,
+  UINT168_ARRAY,
+  UINT176_ARRAY,
+  UINT184_ARRAY,
+  UINT192_ARRAY,
+  UINT200_ARRAY,
+  UINT208_ARRAY,
+  UINT216_ARRAY,
+  UINT224_ARRAY,
+  UINT232_ARRAY,
+  UINT240_ARRAY,
+  UINT248_ARRAY,
+  UINT256_ARRAY,
+  INT8_ARRAY,
+  INT16_ARRAY,
+  INT24_ARRAY,
+  INT32_ARRAY,
+  INT40_ARRAY,
+  INT48_ARRAY,
+  INT56_ARRAY,
+  INT64_ARRAY,
+  INT72_ARRAY,
+  INT80_ARRAY,
+  INT88_ARRAY,
+  INT96_ARRAY,
+  INT104_ARRAY,
+  INT112_ARRAY,
+  INT120_ARRAY,
+  INT128_ARRAY,
+  INT136_ARRAY,
+  INT144_ARRAY,
+  INT152_ARRAY,
+  INT160_ARRAY,
+  INT168_ARRAY,
+  INT176_ARRAY,
+  INT184_ARRAY,
+  INT192_ARRAY,
+  INT200_ARRAY,
+  INT208_ARRAY,
+  INT216_ARRAY,
+  INT224_ARRAY,
+  INT232_ARRAY,
+  INT240_ARRAY,
+  INT248_ARRAY,
+  INT256_ARRAY,
+  BYTES1_ARRAY,
+  BYTES2_ARRAY,
+  BYTES3_ARRAY,
+  BYTES4_ARRAY,
+  BYTES5_ARRAY,
+  BYTES6_ARRAY,
+  BYTES7_ARRAY,
+  BYTES8_ARRAY,
+  BYTES9_ARRAY,
+  BYTES10_ARRAY,
+  BYTES11_ARRAY,
+  BYTES12_ARRAY,
+  BYTES13_ARRAY,
+  BYTES14_ARRAY,
+  BYTES15_ARRAY,
+  BYTES16_ARRAY,
+  BYTES17_ARRAY,
+  BYTES18_ARRAY,
+  BYTES19_ARRAY,
+  BYTES20_ARRAY,
+  BYTES21_ARRAY,
+  BYTES22_ARRAY,
+  BYTES23_ARRAY,
+  BYTES24_ARRAY,
+  BYTES25_ARRAY,
+  BYTES26_ARRAY,
+  BYTES27_ARRAY,
+  BYTES28_ARRAY,
+  BYTES29_ARRAY,
+  BYTES30_ARRAY,
+  BYTES31_ARRAY,
+  BYTES32_ARRAY,
+  BOOL_ARRAY,
+  ADDRESS_ARRAY,
+  BYTES,
+  STRING
 }
 
 /**
  * Get the length of the data for the given schema type
  * (Because Solidity doesn't support constant arrays, we need to use a function)
- * TODO: add more types and make it more efficient (avoid linear search)
  */
 function getStaticByteLength(SchemaType schemaType) pure returns (uint256) {
-  if (schemaType == SchemaType.Uint8) {
-    return 1;
-  } else if (schemaType == SchemaType.Uint16) {
-    return 2;
-  } else if (schemaType == SchemaType.Uint32 || schemaType == SchemaType.Bytes4) {
-    return 4;
-  } else if (schemaType == SchemaType.Uint128) {
-    return 16;
-  } else if (schemaType == SchemaType.Uint256 || schemaType == SchemaType.Bytes32) {
-    return 32;
-  } else if (schemaType == SchemaType.Address) {
-    return 20;
-  } else {
-    // Return 0 for all dynamic types
-    return 0;
+  uint256 index = uint8(schemaType);
+
+  if (index < 32) {
```

Comment:
just noticed we went back from `enum` to hardcoded values here - what's the gas implications of using enums? I feel like enums are much more readable, but on the other hand the enum probably doesn't change anymore (since it already includes all types), so maybe hardcoded numbers are fine if we have test cases to make sure it's always correct

Reply:
~50 gas on each set and get, and since there will be a lot of those it will add up

Reply:
50 gas in relation to how much gas per set in total? 50 out of 10k gas is different than 50 out of 100k gas

Reply:
~2500, but it gets worse for multi-field tables. 1% at best, 5% at worst
(all these numbers are very approximate)

Reply:
imo this optimization is worth it then, as long as we have good test coverage and can reliably detect if anything in this enum changes (adding elements, changing order, etc.)

Reply:
yep, thorough tests are in the next PR (SchemaType by itself isn't meaningful to test, `staticByteLength` is part of the TightCoder suite)

Reply:
agree with this optimization in the context of this being a hot path + the underlying enum is not expected to change + has good test coverage incoming